src/main/java/meteordevelopment/meteorclient/systems/modules/player/ChestSwap.java

```diff
@@ -72,49 +86,45 @@ public void swap() {
 
     private boolean equipChestplate() {
         int bestSlot = -1;
-        boolean breakLoop = false;
-
+        int score = 0;
         for (int i = 0; i < mc.player.getInventory().main.size(); i++) {
-            Item item = mc.player.getInventory().main.get(i).getItem();
+            ItemStack itemStack = mc.player.getInventory().main.get(i);
 
-            switch (chestplate.get()) {
-                case Diamond:
-                    if (item == Items.DIAMOND_CHESTPLATE) {
-                        bestSlot = i;
-                        breakLoop = true;
-                    }
-                    break;
-                case Netherite:
-                    if (item == Items.NETHERITE_CHESTPLATE) {
-                        bestSlot = i;
-                        breakLoop = true;
-                    }
-                    break;
-                case PreferDiamond:
-                    if (item == Items.DIAMOND_CHESTPLATE) {
-                        bestSlot = i;
-                        breakLoop = true;
-                    } else if (item == Items.NETHERITE_CHESTPLATE) {
-                        bestSlot = i;
-                    }
-                    break;
-                case PreferNetherite:
-                    if (item == Items.DIAMOND_CHESTPLATE) {
-                        bestSlot = i;
-                    } else if (item == Items.NETHERITE_CHESTPLATE) {
-                        bestSlot = i;
-                        breakLoop = true;
-                    }
-                    break;
-            }
+            if (chestplate.get().item == itemStack.getItem() || (chestplate.get() == Chestplate.Best && itemStack.getItem() instanceof ArmorItem armor && armor.getSlotType() == EquipmentSlot.CHEST)) {
+                int newScore = getScore(itemStack);
 
-            if (breakLoop) break;
+                if (newScore > score) {
+                    score = newScore;
+                    bestSlot = i;
+                }
+            }
         }
 
         if (bestSlot != -1) equip(bestSlot);
         return bestSlot != -1;
     }
 
+    // modified AutoArmor getScore
+    private int getScore(ItemStack itemStack) {
+        if (itemStack.isEmpty()) return 0;
+
+        int score = 0;
+        Object2IntMap<RegistryEntry<Enchantment>> enchantments = new Object2IntOpenHashMap<>();
```

Comment:
Can you make this map a `private static final` field instead to avoid an allocation. The `Utils.getEnchantments()` method already clears the map.